### ISSUE TITLE
no match for ‘operator==’ 

### ISSUE THREAD

**Issue by thiagotps** [symengine/symengine#1672]
Hi everyone. The following code
``` c++
#include <bits/stdc++.h>
#include <symengine/expression.h>
#include <symengine/functions.h>
#include <symengine/symbol.h>
#include <symengine/add.h>
#include <symengine/mul.h>

using namespace SymEngine;
using namespace std;




int main() {
  const RCP<const Basic> i1 = integer(10);
  const RCP<const Basic> i2 = i1;

  assert(i1 == i2);
  return 0;
}

```
results in the error: 
``` c++
/home/thiago/research/lmscpp/sym.cpp:18:13: error: no match for ‘operator==’ (operand types are ‘const SymEngine::RCP<const SymEngine::Basic>’ and ‘const SymEngine::RCP<const SymEngine::Basic>’)
```

The following code compiles fine:
``` c++
#include <bits/stdc++.h>
#include <symengine/expression.h>
#include <symengine/functions.h>
#include <symengine/symbol.h>
#include <symengine/add.h>
#include <symengine/mul.h>

using namespace SymEngine;
using namespace std;

int main() {
  RCP<const Basic> i1 = integer(10);
  RCP<const Basic> i2 = i1;

  assert(i1 == i2);
  return 0;
}
```

I think the cause for this is the lack of a **const** in the end of ```bool operator==(const RCP<T2> &p2)``` in line 172, file symengine_rcp.h:
```c++
    template <class T2>
    bool operator==(const RCP<T2> &p2)
    {
        return ptr_ == p2.ptr_;
    }
```
That is, i think it should be:
```c++
    template <class T2>
    bool operator==(const RCP<T2> &p2) const
    {
        return ptr_ == p2.ptr_;
    }
```
Is this a bug ? Or its me using ```RCP``` wrongly ? Thanks in advance.

**Comment by isuruf** [comment]
Yes, it should be marked const. Btw, `==` will compare pointer equality and not that the two objects are equal. i.e. `i1 == integer(10)` will return False as they are two different Integer objects with the same value 10. You should use `eq(*i1, *i2)` if you want to check that the objects are equal.

**Comment by naveensaigit** [comment]
> Yes, it should be marked const.

Is this change required? I can add it

**Comment by lkeegan** [comment]
fixed in #1756
